### ISSUE TITLE
Adding a project fails due to system policy error (15-2213)

### ISSUE THREAD

**Issue by steveblamey** [uzerpllp/uzerp#10]
When adding a project if no Key contact is defined the project adds OK, conversely if a key contact IS defined the project will not be added and a blank screen appears. This manifested itself while trying to add projects in the Oakmount test instance.

From Dave Easeman 24/4/14
The problem was related to policies; in this case it was trying to add a policy constraint due to a fk defined on person to company but was failing because there are no policies defined in llp so policies are disabled. In other words, because policies are not defined in llp, and are therefore disabled, it should not try to add a policy constraint due to a fk definition.

Dave directly amended uzerpllp and base instances - this means that the fix is in the 2014.3 deb and therefore the master branch. However it may not have been propogated to all of the instances.

**Comment by steveblamey** [comment]
Dave has fixed this in 'base' (which is used to create the deb file) and the CVS repo on the hosted server. The code changes are detailed below lib/classes/standard/DataObject.php

line 604

```
#!php

         if (!SYSTEM_POLICIES_ENABLED)
         {
             return;
         }
```


line 611

```
#!php


             if (!($this->_policyConstraint['constraint'] instanceof ConstraintChain))
             {
                 $this->_policyConstraint['constraint'] = new ConstraintChain();
             }
```



After the change, it should look like this:-

```
#!php


                if (!SYSTEM_POLICIES_ENABLED)
                {
                        return;
                }

                if ($cc instanceof ConstraintChain && $cc->count()>0)
                {

                        if (!($this->_policyConstraint['constraint'] instanceof ConstraintChain))
                        {
$this->_policyConstraint['constraint'] = new ConstraintChain();
                        }

$this->_policyConstraint['constraint']->add($cc, 'OR');
                }

```

**Comment by steveblamey** [comment]
Also the DataObject.php version number at line 11 needs to change to 1.133.

**Comment by steveblamey** [comment]
fixes #10 (uzerp ticket 15-2213)
Adding a project fails due to system policy error

Requested code already applied to 2013.4, but changed file version.


→ <<cset e30b2c3e332f>>
